homeworks/07-mars-viewer/src/features/pages/sols/components/SolSelector/SolSelector.tsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import styles from './SolSelector.module.css';
+
+interface SolSelectorProps {
+  readonly sol: number;
+  readonly selectSol: (sol: number) => void;
+  readonly loadSol: () => void;
+}
+
+const SolSelector = ({ sol, selectSol, loadSol }: SolSelectorProps) => {
+  const changeSol = (event: React.ChangeEvent<HTMLInputElement>) => {
+    selectSol(+event.target.value);
+  };
+  return (
+    <div className={styles.SolSelector}>
+      <input
+        data-testid="solInput"
+        min="0"
+        defaultValue={sol}
+        type="number"
+        onChange={changeSol}
+      />
+      <button onClick={loadSol}>Load</button>
```

Comment:
In example app http://mars-viewer.surge.sh/ button being disabled based on request state while it's pending, so user can not make a lot of requests by quickly clicking button